From c7ca6e38fad70bd335c552c9cee86ce8fa1a0502 Mon Sep 17 00:00:00 2001
From: Pau Ruiz Safont <pau.safont@vates.tech>
Date: Wed, 17 Jun 2026 16:11:31 +0100
Subject: [PATCH] quicktest: test for number of CD VBDs

This shows the current code leaks the VBDs

Signed-off-by: Pau Ruiz Safont <pau.safont@vates.tech>
(cherry picked from commit 38cfbc549040b68f85d50e963546864ac2606a40)
---
 ocaml/quicktest/quicktest_vm_snapshot.ml | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ocaml/quicktest/quicktest_vm_snapshot.ml b/ocaml/quicktest/quicktest_vm_snapshot.ml
index 6463922b7..0279f95d5 100644
--- a/ocaml/quicktest/quicktest_vm_snapshot.ml
+++ b/ocaml/quicktest/quicktest_vm_snapshot.ml
@@ -162,12 +162,17 @@ let test_revert rpc session_id vm vdi vdi2 ~change =
 
 let test_revert_cds rpc session_id vm vdi vdi2 =
   let snapshot = take_snapshot rpc session_id vm ~origin:__FUNCTION__ in
+
+  let snap_vbds = Client.Client.VM.get_VBDs ~rpc ~session_id ~self:snapshot in
+  Alcotest.(check int) "Snapshot must only have 2 VBDs" 2 (List.length snap_vbds) ;
+
   Client.Client.VM.revert ~rpc ~session_id ~snapshot ;
 
   let vbds = Client.Client.VM.get_VBDs ~rpc ~session_id ~self:vm in
   let vdi_after = get_vdi_with_user_device rpc session_id vbds "0" in
   let vdi_after2 = get_vdi_with_user_device rpc session_id vbds "1" in
 
+  Alcotest.(check int) "VM must only have 2 VBDs" 2 (List.length vbds) ;
   (* CD VDIs are considered immutable and the clone code ignores them *)
   check_vdis_same vdi vdi_after ;
   check_vdis_same vdi2 vdi_after2
